src/Catalyst.Core.Modules.Dfs/UnixFs/UnixFs.cs

```diff
@@ -0,0 +1,130 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs.CoreApi;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Abstractions.Keystore;
+using Catalyst.Core.Lib.Dag;
+using Lib.P2P;
+using ProtoBuf;
+
+namespace Catalyst.Core.Modules.Dfs.UnixFileSystem
+{
+    /// <summary>
+    ///   Support for the *nix file system.
+    /// </summary>
+    public class UnixFs
+    {
+        static readonly byte[] emptyData = new byte[0];
+
+        /// <summary>
+        ///   Creates a stream that can read the supplied <see cref="Cid"/>.
+        /// </summary>
+        /// <param name="id">
+        ///   The identifier of some content.
+        /// </param>
+        /// <param name="blockService">
+        ///   The source of the cid's data.
+        /// </param>
+        /// <param name="keyChain">
+        ///   Used to decypt the protected data blocks.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation. The task's value is
+        ///   a <see cref="Stream"/> that produces the content of the <paramref name="id"/>.
+        /// </returns>
+        /// <remarks>
+        ///  The id's <see cref="Cid.ContentType"/> is used to determine how to read
+        ///  the conent.
+        /// </remarks>
+        public static Task<Stream> CreateReadStreamAsync(Cid id,
+            IBlockApi blockService,
+            IKeyApi keyChain,
+            CancellationToken cancel)
+        {
+            // TODO: A content-type registry should be used.
+            if (id.ContentType == "dag-pb")
+                return CreateDagProtoBufStreamAsync(id, blockService, keyChain, cancel);
+            else if (id.ContentType == "raw")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)